modules/ensemble/lib/widget/inline_timepicker.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:intl/intl.dart'; // For date formatting
+import '../framework/event.dart';
+
+/// An inline time picker widget that mimics the iOS clock app's circling time picker.
+/// It provides various customization options for a seamless user experience.
+class InlineTimePicker extends StatefulWidget
+    with
+        Invokable,
+        HasController<InlineTimePickerController, InlineTimePickerState> {
+  static const type = 'InlineTimePicker';
+
+  InlineTimePicker({Key? key}) : super(key: key);
+
+  final InlineTimePickerController _controller = InlineTimePickerController();
+
+  @override
+  get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => InlineTimePickerState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'selectedTime': () => _controller.selectedTimeFormatted(),
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {
+      'reset': () => _controller.reset(),
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'initialTime': (time) =>
+          _controller.initialTime = Utils.getTimeOfDay(time),
+      'onTimeChanged': (funcDefinition) => _controller.onTimeChangedAction =
+          ensemble.EnsembleAction.from(funcDefinition, initiator: this),
+      'mode': (value) => _controller.mode =
+          Utils.getEnum<CupertinoTimerPickerMode>(
+              value, CupertinoTimerPickerMode.values),
+      'minuteInterval': (value) =>
+          _controller.minuteInterval = Utils.optionalInt(value),
+      'secondInterval': (value) =>
+          _controller.secondInterval = Utils.optionalInt(value),
+      'onTimeChangedHaptic': (value) =>
+          _controller.onTimeChangedHaptic = Utils.optionalString(value),
+    };
+  }
+}
+
+class InlineTimePickerController extends BoxController {
+  TimeOfDay? initialTime;
+  TimeOfDay? selectedTime;
+  int selectedSeconds = 0; // To hold the selected seconds
+  ensemble.EnsembleAction? onTimeChangedAction;
+  CupertinoTimerPickerMode mode = CupertinoTimerPickerMode.hm;
+  int? minuteInterval;
+  int? secondInterval;
+  String? onTimeChangedHaptic;
+
+  /// Resets the time picker to the initial time.
+  void reset() {
+    selectedTime = initialTime;
+    selectedSeconds = 0;
+  }
+
+  /// Returns the selected time formatted as a string without requiring context.
+  String selectedTimeFormatted() {
+    if (selectedTime == null) {
+      return '';
+    }
+
+    final now = DateTime.now();
+    final formattedTime = DateFormat('hh:mm:ss a').format(DateTime(
+      now.year,
+      now.month,
+      now.day,
+      selectedTime!.hour,
+      selectedTime!.minute,
+      selectedSeconds,
+    ));
+
+    return formattedTime;
+  }
+}
+
+class InlineTimePickerState extends EWidgetState<InlineTimePicker> {
+  @override
+  void initState() {
+    super.initState();
+    widget._controller.selectedTime =
+        widget._controller.initialTime ?? TimeOfDay.now();
```

Comment:
why not just do widget.controller? what's the point of accessing the private property?